### ISSUE TITLE
LDSearch should behave more similarly to web's SNAP in regards to query SNPs that are missing in dataset.

### ISSUE THREAD

**Issue by ramiromagno** [ropensci/rsnps#37]
`LDSearch` should behave more similarly to [SNAP](http://archive.broadinstitute.org/mpg/snap/ldsearch.php) in regards to query SNPs that are missing in dataset.

Example: 
```
ld_search("rs11571833", dataset = "onekgpilot", panel = "CEU", RSquaredLimit = 0.8,
          distanceLimit = 500, GeneCruiser = FALSE, quiet = FALSE)
```

outputs

```
Querying SNAP...
Error: no valid data found
In addition: Warning messages:
1: rs11571833	WARNING	Query snp not in 1000GenomesPilot1 
2: rs11571833	WARNING	No matching proxy snps found 
```

In my opinion, the first issue is that this behaves differently from the website. [SNAP](http://archive.broadinstitute.org/mpg/snap/ldsearch.php) will not break if among the provided list of query SNPs some are missing from the queried dataset. 
The second problem is that this breaks the computation of `LDSearch` when calling it when on a vector of several query SNPs.

Therefore, I think Broad's behavior should be the default behavior of `LDSearch`. Perhaps the returned list by `LDSearch` should see the data frames corresponding to the offending query SNPs be removed by assigning to them `NULL` (in a list this will remove them). In addition, one should add one argument to `LDSearch` to switch between the old and this new proposed behavior, making the latter the default option.

**Comment by sckott** [comment]
@ramiromagno did you mean to close this?

**Comment by ramiromagno** [comment]
@sckott Yes, I realized this "issue" was more of a "wish feature"...  Let me explain. I am collaborating with a colleague who is using this package. At a certain point she was getting those warnings related with issue #34. Without paying too much attention, we mistook the warning for something it was not. We thought that  `LDSearch` would not take a vector of SNPs, but one SNP at a time. So we immediately wrote a wrapper around `LDSearch` with lapply to run `LDSearch` over a vector of SNPs. However, this check inside `LDSearch`'s code:

```
 # check whether any real data or not
  if (all(unlist(out_split$SNP[1,], use.names = FALSE) %in% names(out_split$SNP)) &&
      NROW(out_split$SNP) == 1) {
    stop("no valid data found", call. = FALSE)
  }
```
 
leads to a funny consequence, which is, if we're using `LDSearch` on one SNP at a time, inside an `lapply` loop for nonexistent query SNPs, `LDSearch` will `stop`. As it stops, it breaks the `lapply` loop, hence breaking the whole computation over the SNPs... :D This does not happen when we pass a vector of SNPs to `LDSearch` as it nicely skips those missing query SNPs, unless they are all missing, in which case, stopping the computation is the sensible thing to do.

In other words, one could say that `LDSearch` is "vectorizable " via the apply family of functions because it will stop on missing query SNP, which is probably undesirable. Nevertheless, taking this point of view, refactoring the function to work in this case is more a of a feature request than a bug.

Yet, now that I am looking at your code of that check, it does not seem right indeed... Did you not want to use `out` instead of `out_split` in the above-mentioned code?
